internal/projects/deleter.go

```diff
@@ -170,3 +179,44 @@ func hasOtherRoleAssignments(as []*v1.RoleAssignment, subject string) bool {
 		return a.GetRole() == authz.RoleAdmin.String() && a.Subject != subject
 	})
 }
+
+func exportProjectMetadata(ctx context.Context, projectID uuid.UUID, qtx db.Querier) (*logger.ProjectTombstone, error) {
+	var (
+		profilesCount int64
+		reposCount    int64
+		entitlements  []db.Entitlement
+	)
+
+	g, ctx := errgroup.WithContext(ctx)
+	g.Go(func() error {
+		var err error
+		profilesCount, err = qtx.CountProfilesByProjectID(ctx, projectID)
+		return err
+	})
+	g.Go(func() error {
+		var err error
+		reposCount, err = qtx.CountRepositoriesByProjectID(ctx, projectID)
+		return err
+	})
+	g.Go(func() error {
+		var err error
+		entitlements, err = qtx.GetEntitlementsByProjectID(ctx, projectID)
+		return err
+	})
+
+	if err := g.Wait(); err != nil {
+		return nil, fmt.Errorf("error getting project metadata: %w", err)
+	}
+
+	var entitlementsFeatures []string
```

Comment:
Use `make` with a size argument to avoid needing extra resizes:

```suggestion
	entitlementsFeatures := make([]string, 0, len(entitlements))
```